companyweb_base/models/partner_cweb_model.py

```diff
@@ -0,0 +1,527 @@
+            else:
+                self.cweb_creditLimit_enable = False
+            if cweb_response["CreditLimit"]["Value"]["Info"]:
+                self.cweb_creditLimit_info = cweb_response["CreditLimit"]["Value"][
+                    "Info"
+                ]
+                self.cweb_creditLimit_info_enable = True
+            else:
+                self.cweb_creditLimit_info_enable = False
+        else:
+            self.cweb_creditLimit = None
+            self.cweb_creditLimit_info = None
+            self.cweb_creditLimit_info_enable = False
+
+        self.cweb_warnings_enable = cweb_response["WarningsOverview"]["IsEnabled"]
+        cweb_has_warnings_values = cweb_response["WarningsOverview"]["Value"]
+        separator = "\n"
+        if self.cweb_warnings_enable and cweb_has_warnings_values:
+            if cweb_response["WarningsOverview"]["Value"]["Warnings"]:
+                self.cweb_warnings = separator.join(
+                    cweb_response["WarningsOverview"]["Value"]["Warnings"]["string"]
+                )
+            else:
+                self.cweb_warnings_enable = False
+        else:
+            self.cweb_warnings = None
+
+    def button_companyweb_enhance(self):
+        """Main logic of the module
+        Validate that the logged in user has credentials
+        make the API CALL
+        based on status make decision
+        When status is ok -> populate fields"""
+        user_login = self.env.user.cweb_login
+        user_password = self.env.user.cweb_password
+        if not user_login or not user_password:
+            return self._call_wizard_credentials("No Companyweb credentials")
+
+        client = zeep.Client("https://connect.companyweb.be/V1.3/alacarteservice.asmx")
+        r = client.service.GetCompanyByVat(
+            dict(
+                CompanyWebLogin=user_login,
+                CompanyWebPassword=user_password,
+                ServiceIntegrator=SERVICE_INTEGRATOR_ID,
+                LoginHash=self._create_hash(
+                    user_login, user_password, SERVICE_INTEGRATOR_SECRET
+                ),
+                Language="FR",
+                VatNumber=self.vat,
+            )
+        )
+        if r["StatusCode"] in [101, 302]:
+            return self._call_wizard_credentials("Bad Companyweb credentials")
+        elif r["StatusCode"] != 0:
+            raise UserError(
+                _("Companyweb status : %s : %s ", r["StatusCode"], r["StatusMessage"])
+            )
+
+        cweb_response = r["CompanyResponse"]
+        self._populate_general_cweb(cweb_response)
+        self._populate_address_cweb(cweb_response)
+        self._populate_balans_cweb(cweb_response)
+        self._populate_url_cweb(cweb_response)
+        self._populate_dates_cweb(cweb_response)
+        self._populate_score_cweb(cweb_response)
+        self._populate_cweb_data(cweb_response)
+        # self._set_default_value()
+
+    def button_companyweb_address(self):
```

Comment:
```suggestion
    def button_companyweb_copy_address(self):
```

Reply:
method changed to prefix with cweb 
cweb_button_enhance
cweb_button_copy_address